aws-sns-subscription/src/main/java/software/amazon/sns/subscription/ListHandler.java

```diff
@@ -35,34 +21,25 @@ public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final ListSubscriptionsByTopicRequest listSubscriptionsByTopicRequest = Translator.translateToListSubscriptionsByTopicRequest(request);
-
-        final ListSubscriptionsByTopicResponse listSubscriptionsByTopicResponse;
-        try {
-               listSubscriptionsByTopicResponse = proxy.injectCredentialsAndInvokeV2(listSubscriptionsByTopicRequest, proxyClient.client()::listSubscriptionsByTopic);
+        this.logger = logger;
+        ResourceModel resourceModel = request.getDesiredResourceState();
 
-        } catch (final SubscriptionLimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
-        } catch (final FilterPolicyLimitExceededException e) {
-            throw new CfnServiceLimitExceededException(e);
-        } catch (final InvalidParameterException e) {
-            throw new CfnInvalidRequestException(e);
-        } catch (final InternalErrorException e) {
-            throw new CfnInternalFailureException(e);
-        } catch (final NotFoundException e) {
-            throw new CfnNotFoundException(e);
-        } catch (final AuthorizationErrorException e) {
-            throw new CfnAccessDeniedException(e);
-        } catch (final InvalidSecurityException e) {
-            throw new CfnInvalidCredentialsException(e);
+        if (resourceModel == null) {
+            return ProgressEvent.failed(resourceModel, callbackContext, HandlerErrorCode.InvalidRequest, "List request is invalid");
         }
-        final List<ResourceModel> models = Translator.translateFromListRequest(listSubscriptionsByTopicResponse);
 
-        return ProgressEvent.<ResourceModel, CallbackContext>builder()
-                .resourceModels(models)
-                .nextToken(listSubscriptionsByTopicResponse.nextToken())
-                .status(OperationStatus.SUCCESS)
-                .build();
+        logger.log(String.format("[StackId: %s, ClientRequestToken: %s] Calling List SNS Subscription", request.getStackId(),
+                request.getClientRequestToken()));
+
+        return proxy.initiate("AWS-SNS-Subscription::List", proxyClient, resourceModel, callbackContext)
+                .translateToServiceRequest(m -> Translator.translateToListSubscriptionsRequest(request.getNextToken()))
+                .makeServiceCall((getSubscriptionsRequest, client) -> client.injectCredentialsAndInvokeV2(getSubscriptionsRequest, client.client()::listSubscriptions))
+                .handleError((awsRequest, exception, client, model, context) -> handleError(awsRequest, exception, client, model, context))
+                .done((getSubscriptionsRequest, getSubscriptionsResponse, client, resourceModel1, context) ->
+                        ProgressEvent.<ResourceModel, CallbackContext>builder()
+                                .resourceModels(Translator.translateFromListRequest(getSubscriptionsResponse))
```

Comment:
List handler should only return the primaryIds I see topicArn & protocol properties in your translateFromListRequest model